nautobot/core/filters.py

```diff
@@ -386,6 +386,7 @@ def filter(self, qs, value):
         return qs.distinct()
 
 
+# TODO(timizuo): NaturalKeyOrPKMultipleChoiceFilter is not currently handling pk Integer field properly
```

Comment:
Please open an issue if there isn't one already, and reference it by number here and below.